src/backend/executor/execMain.c

```diff
@@ -4091,6 +4091,7 @@ EvalPlanQualSetTuple(EPQState *epqstate, Index rti, HeapTuple tuple)
 {
 	EState	   *estate = epqstate->estate;
 
+	Insist(estate);
```

Comment:
what about EvalPlanQualGetTuple below?
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/nodeLockRows.c#L267

Reply:
Added a check

Reply:
I think the [epq_started](https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/nodeLockRows.c#L43) variable can be replaced by checking `node->lr_epqstate.estate != NULL` because they mean the same thing.

Reply:
Can you explain why? It's not clear to me

Reply:
I think we can replace this code https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/executor/nodeLockRows.c#L84-L86 by
```c
		/* clear any leftover test tuple for this rel */
		if (epq_started)
		{
			Assert(node->lr_epqstate.estate != NULL);
			EvalPlanQualSetTuple(&node->lr_epqstate, erm->rti, NULL);
		}
```

Reply:
It's not the line we got problem with

Reply:
we will replace condition `if (node->lr_epqstate.estate != NULL)` by assert and static analyzer will not see it

Reply:
Done